### PR TITLE
SW-1749 Split SearchServiceTest

### PR DESCRIPTION
**Pull request by charlie-tf**
- Create abstract base class for SearchServiceTest
  - Inherits DatabaseTest
  - Implements RunAsUser
  - Contains helper vals/funs
- Divide tests by purpose

### CODE REVIEW

File: src/test/kotlin/com/terraformation/backend/seedbank/search/SearchServiceBasicSearchTest.kt [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import com.terraformation.backend.search.NoConditionNode
+import com.terraformation.backend.search.SearchDirection
+import com.terraformation.backend.search.SearchFilterType
+import com.terraformation.backend.search.SearchResults
+import com.terraformation.backend.search.SearchSortField
+import io.mockk.every
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+internal class SearchServiceBasicSearchTest : SearchServiceTest() {
```
Comment (charlie-tf) [inline src/test/kotlin/com/terraformation/backend/seedbank/search/SearchServiceBasicSearchTest.kt:18]: One area I still questions a little bit is this class. These are the "leftover" tests after I split out the other classes. They seems to exercise the basic search facilities of the application, for the most part, which was the reason for the name.

Comment (sgrimm) [reply src/test/kotlin/com/terraformation/backend/seedbank/search/SearchServiceBasicSearchTest.kt:18]: Yeah, this seems fine.

**Review by sgrimm — APPROVED** [review]
Terrific. This was a real beast of a test class.

**Comment by sgrimm** [comment]
I have a couple other merges queued up in Graphite right now but I'll merge this one when they're done.

**Comment by charlie-tf** [comment]
> Terrific. This was a real beast of a test class.

It was a big'un!

**Pull request merged**